docs/getting-started.md

```diff
@@ -0,0 +1,260 @@
+  ```solidity
+  function check_transfer() {
+      address sender = svm.createAddress("sender");
+      address receiver = svm.createAddress("receiver");
+      uint256 amount = svm.createUint256("amount");
+      ...
+  }
+  ```
+
+- Halmos requires dynamically-sized arrays (including `bytes` and `string`) to be given with a fixed size. Thus they cannot be declared as input parameters, but need to be programmatically constructed. For example, a byte array can be generated using the `svm.createBytes()` cheatcode as follows:
+  ```solidity
+  bytes memory data = svm.createBytes(96, 'data');
+  ```
+  Similarly, a dynamic array of integers can be created as shown below:
+  ```solidity
+  uint256[] memory arr = new uint256[3];
+  for (uint i = 0; i < 3; i++) {
+      arr[i] = svm.createUint256('element');
+  }
+  ```
+  We are planning to implement more cheatcodes and features that can make it easier to declare or create dynamic arrays.
+
+### 2.2 Specify input conditions
+
+Recall that symbolic tests take into account all possible input combinations. However, not all input combinations are relevant or valid for every test scenario. Similar to fuzz tests, you can use `vm.assume()` to specify the conditions for valid inputs.
+
+In our example, the conditions for the valid sender and receiver addresses are specified as follows:
+```solidity
+vm.assume(receiver != address(0));
+vm.assume(token.balance(sender) >= amount);
+```
+Like fuzz tests, any input combinations that don't satisfy the `assume()` conditions are disregarded. This means that, after executing the above `assume()` statements, only the input combinations in which the receiver is non-zero and the sender has sufficient balance are considered. Other input combinations that violate these conditions are ignored.
+
+**Tips:**
+
+- You need to be careful not to exclude valid inputs by setting too strong input conditions.
+
+- In symbolic tests, avoid using `bound()` as it tends to perform poorly. Instead, use `vm.assume()` which is more efficient and easier to read and write.
+
+### 2.3 Call target contracts
+
+Now you can invoke the target contracts with the prepared input symbols.
+
+In our example, the transfer function is called with the symbolic receiver address and amount. The `prank()` cheatcode is also used to set `msg.sender` to the symbolic sender address, as shown below:
+```solidity
+vm.prank(sender);
+token.transfer(receiver, amount);
+```
+
+**Tips:**
+
+- If your goal is to check whether the target contract reverts under the expected conditions, a low-level call should be used. This allows the execution to continue even if the external call fails. Below is an example of a low-level call to the token transfer function. Note that the return value `success` can be subsequently used to check the reverting conditions.
+  ```solidity
+  vm.prank(sender);
+  (bool success,) = address(token).call(
+      abi.encodeWithSelector(token.transfer.selector, receiver, amount)
+  );
```

Comment:
do you want to add `assert(!success)` in the example?

that's something we could probably be doing by implementing `vm.expectRevert()`, right?